src/Cli/dotnet/commands/dotnet-workload/GlobalJsonWorkloadSetFile.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+namespace Microsoft.DotNet.Workloads.Workload
+{
+    internal class GlobalJsonWorkloadSetsFile
+    {
+        string _path;
+
+        private static JsonSerializerOptions _jsonSerializerOptions = new JsonSerializerOptions()
+        {
+            WriteIndented = true,
+            DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull
+        };
+
+        public GlobalJsonWorkloadSetsFile(SdkFeatureBand sdkFeatureBand, string dotnetDir)
+        {
+            _path = Path.Combine(WorkloadInstallType.GetInstallStateFolder(sdkFeatureBand, dotnetDir), "globaljsonworkloadsets.json");
+        }
+
+        public void RecordWorkloadSetInGlobalJson(string globalJsonPath, string workloadSetVersion)
+        {
+            //  Create install state folder if needed
+            Directory.CreateDirectory(Path.GetDirectoryName(_path));
```

Comment:
Where is this being created? If we're creating files for global install, it needs to go through the proper APIs to configure ACLs

Reply:
Good catch, I've fixed this to apply the ACLs.